{{cookiecutter.profile_name}}/lsf_submit.py

```diff
@@ -0,0 +1,187 @@
+        return self.job_properties.get("wildcards", dict())
+
+    @property
+    def wildcards_str(self):
+        return ".".join("{}={}".format(k, v) for k, v in self.wildcards.items()) or "unique"
+
+    @property
+    def rule_name(self):
+        return self.job_properties.get("rule", "rule_name")
+
+    @property
+    def groupid(self):
+        return self.job_properties.get("groupid", "group")
+
+    @property
+    def is_group_jobtype(self):
+        return self.job_properties.get("type", "") == "group"
+
+    @property
+    def jobname(self) -> str:
+        if self.is_group_jobtype:
+            return "{groupid}_{jobid}".format(groupid=self.groupid, jobid=self.jobid)
+        else:
+            return self.cluster.get("jobname",
+                                    "{rule_name}.{wildcards_str}".format(rule_name=self.rule_name,
+                                                                         wildcards_str=self.wildcards_str))
+
+    @property
+    def jobid(self):
+        if self.is_group_jobtype:
+            return self.job_properties.get("jobid", "").split("-")[0]
+        else:
+            return self.job_properties.get("jobid")
+
+    @property
+    def logdir(self):
+        return Path(self.cluster.get("logdir", CookieCutter.get_log_dir()))
+
+    @property
+    def outlog(self):
+        return self.logdir / "{jobid}_{random_string}.out".format(jobid=self.jobid, random_string=self.random_string)
+
+    @property
+    def errlog(self):
+        return self.logdir / "{jobid}_{random_string}.err".format(jobid=self.jobid, random_string=self.random_string)
+
+    @property
+    def jobinfo_cmd(self):
+        return '-o "{out_log}" -e "{err_log}" -J "{jobname}"'.format(out_log=self.outlog, err_log=self.errlog, jobname=self.jobname)
+
+    @property
+    def queue(self):
+        return self.cluster.get("queue", "")
+
+    @property
+    def queue_cmd(self):
+        return "-q {}".format(self.queue) if self.queue else ""
+
+    @property
+    def cluster_cmd(self):
+        return self._cluster_cmd
+
+    @property
+    def submit_cmd(self):
+        return "bsub {resources} {job_info} {queue} {cluster} {jobscript}".format(
+                    resources=self.resources_cmd,
+                    job_info=self.jobinfo_cmd,
+                    queue=self.queue_cmd,
+                    cluster=self.cluster_cmd,
+                    jobscript=self.jobscript,
+        )
+
+    def _create_logdir(self):
+        OSLayer.mkdir(self.logdir)
+
+    def _remove_previous_logs(self):
+        OSLayer.remove_file(self.outlog)
+        OSLayer.remove_file(self.errlog)
+
+    def _submit_cmd_and_get_external_job_id(self):
+        output_stream, error_stream = OSLayer.run_process_and_get_output_and_error_stream(self.submit_cmd)
+        match = re.search(r"Job <(\d+)> is submitted", output_stream)
+        jobid = match.group(1)
+        return int(jobid)
+
+    def _get_information_to_status_script(self, external_job_id: int):
```

Comment:
I know it's annoying but could you please add type annotations?

Reply:
The name of this function is confusing. I don't get what it is doing?

Reply:
I refactored the name of this function to `_get_parameters_to_status_script`. What is returned by this function is the command-line parameters/arguments that the status script (`lsf_status.py`) will receive. Implemented at https://github.com/leoisl/snakemake-lsf/commit/53402769805b3bdccc131bc455996ebec3a9092a